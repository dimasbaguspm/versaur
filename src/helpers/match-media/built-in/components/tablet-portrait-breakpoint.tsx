import { BREAKPOINT_TABLET_PORTRAIT } from '../../constants'
import { MatchMedia } from '../../match-media'

/**
 * Props for breakpoint components
 */
interface BreakpointComponentProps {
  children: React.ReactNode
}

/**
 * Component that only renders children on tablet portrait viewports
 *
 * @example
 * ```tsx
 * <TabletPortraitBreakpoint>
 *   <TabletPortraitContent />
 * </TabletPortraitBreakpoint>
 * ```
 */
export function TabletPortraitBreakpoint({
  children,
}: BreakpointComponentProps) {
  return <MatchMedia query={BREAKPOINT_TABLET_PORTRAIT}>{children}</MatchMedia>
}
