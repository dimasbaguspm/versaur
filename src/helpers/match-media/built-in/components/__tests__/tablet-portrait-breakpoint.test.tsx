import { render, screen } from '@testing-library/react'
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest'
import { TabletPortraitBreakpoint } from '../tablet-portrait-breakpoint'
import {
  createMatchMediaMock,
  setupMatchMediaMock,
} from '../../../test-helpers'

describe('TabletPortraitBreakpoint', () => {
  const { beforeEach: setupBefore, afterEach: setupAfter } =
    setupMatchMediaMock()

  beforeEach(() => {
    setupBefore()
  })

  afterEach(() => {
    setupAfter()
  })

  it('should render children when viewport is tablet portrait', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <TabletPortraitBreakpoint>
        <div data-testid='tablet-portrait-content'>Tablet Portrait Content</div>
      </TabletPortraitBreakpoint>
    )

    expect(screen.getByTestId('tablet-portrait-content')).toBeInTheDocument()
  })

  it('should not render children when viewport is not tablet portrait', () => {
    const mockMatchMedia = createMatchMediaMock(false)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <TabletPortraitBreakpoint>
        <div data-testid='tablet-portrait-content'>Tablet Portrait Content</div>
      </TabletPortraitBreakpoint>
    )

    expect(
      screen.queryByTestId('tablet-portrait-content')
    ).not.toBeInTheDocument()
  })

  it('should match snapshot when rendered', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    const { asFragment } = render(
      <TabletPortraitBreakpoint>
        <div>Tablet Portrait Content</div>
      </TabletPortraitBreakpoint>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
