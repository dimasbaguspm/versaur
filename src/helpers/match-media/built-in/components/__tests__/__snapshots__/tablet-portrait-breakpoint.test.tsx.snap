// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabletPortraitBreakpoint > should match snapshot when rendered 1`] = `
<DocumentFragment>
  <div>
    Tablet Portrait Content
  </div>
</DocumentFragment>
`;
